kratos/geometries/geometry.h

```diff
@@ -1122,41 +1122,102 @@ class Geometry : public PointerVector<TPointType>
         return false;
     }
 
+    ///@}
+    ///@name Boundaries
+    ///@{
+
+    /**
+     * @brief This method gives you all boundaries entities of this geometry.
+     * @details This method will gives you all the boundaries entities
+     * @return GeometriesArrayType containes this geometry boundaries entities.
+     * @see GeneratePoints()
+     * @see GenerateEdges()
+     * @see GenerateFaces()
+     */
+    virtual GeometriesArrayType GenerateBoundariesEntities( void )
+    {
+        const SizeType dimension = this->LocalSpaceDimension();
+        if (dimension == 3) {
+            return this->GenerateFaces();
+        } else if (dimension == 2) {
+            return this->GenerateEdges();
+        } else if (dimension == 1) {
+            return this->GeneratePoints();
+        }
+
+        return GeometriesArrayType();
+    }
+
+    ///@}
+    ///@name Points
+    ///@{
+
+    /**
+     * @brief This method gives you all points of this geometry.
+     * @details This method will gives you all the points
+     * @return GeometriesArrayType containes this geometry points.
+     * @see Points()
+     */
+    virtual GeometriesArrayType GeneratePoints( void )
+    {
+        GeometriesArrayType points;
+
+        const auto& p_points = this->Points();
+        for (IndexType i_point = 0; i_point < p_points.size(); ++i_point) {
+            PointsArrayType point_array;
+            point_array.push_back(p_points(i_point));
+            auto p_point_geometry = Kratos::make_shared<GeometryType>(point_array, mpGeometryData);
```

Comment:
why are u passing mpGeometryData here? i don't think it is correct...

Reply:
Because I need to build a geometry, I cannot include a point as far as point depends on geometry, so it is a generic geometry of only one node, which requires in the constructor to pass mpGeometryData (it does not compile without it, I tried)

Reply:
could u try if it compiles as:

~~~cpp
auto p_point_geometry = Kratos::make_shared<Geometry<TPointType>(point_array);
~~~

i had difficulties with GeometryType before